build.ps1

```diff
@@ -0,0 +1,290 @@
+function BuildPyInstaller() {
+    $req = Get-Content requirements.txt
+    $imports = @('glcontext', 'openimageio', 'wx', 'numpy', 'opencv-python', 'gswidgetkit', 'gsnodegraph')
+    $tmp = $env:TEMP
+
+    $installer = 'pyinstaller'
+    $args = [System.Collections.ArrayList]@()
+    # [void]$args.Add('--upx-dir .\upx')
+    [void]$args.Add("${installer} ")
+    # [void]$args.Add("--onefile ")
+    [void]$args.Add("--noconfirm")
+    [void]$args.Add("-n GimelStudio")
+    [void]$args.Add("$(PyInstallAddData)")
+    [void]$args.Add("--workpath '${tmp}'")
+    [void]$args.Add("--runtime-tmpdir ${tmp}")
+    [void]$args.Add("--exclude-module=.git")
+    [void]$args.Add("-i assets/GIMELSTUDIO_ICO.ico")
+    [void]$args.Add("$($req | % { "--hidden-import $_".Trim() })")
+    [void]$args.Add("$($imports | % { "--hidden-import $_".Trim() })")
+    [void]$args.Add("src/main.py")
+    return $args
+}
+
+# --| Nuitka -----------------------------------------
+function BuildNuitka() {
+    # -- Install ccache to greatly speed --------
+    # -- up subsequent compilations -------------
+    try {
+        $ccachePath = [System.IO.Path]::Combine($currentPath, '.tools', 'ccache')
+        $ccacheUrl = "https://github.com/ccache/ccache/releases/download/v4.5.1/ccache-4.5.1-windows-64.zip"
+
+        if (!(test-path -path $ccachePath)) {
+            mkdir $ccachePath -force | out-null
+
+            $progressPreference = 'silentlyContinue'
+            $response = invoke-webrequest $ccacheUrl -OutFile "${ccachePath}/ccache.zip"
+            $progressPreference = 'Continue'
+            Expand-Archive -Path "${ccachePath}/ccache.zip" -DestinationPath $ccachePath -force
+        }
+
+        $ccacheBin = $ccachePath | gci -recurse | where { $_.Extension -eq '.exe' }
+        if ( [string]::IsNullOrEmpty($ccacheBin[0].FullName)) {
+            echo "Unable to use ccache, continuing regardless..."
+        }
+        else { $env:NUITKA_CCACHE_BINARY = $ccacheBin[0].FullName; $env:PATH = $env:PYTHONPATH += $ccacheBin[0].Directory }
+    }
+    catch {
+        echo "Unable to use ccache, continuing regardless..."
+    }
+
+    # -- Nuitka build settings ------------------
+    $installer = 'nuitka'
+    $args = [System.Collections.ArrayList]@()
+    [void]$args.Add("python -m ${installer} ")
+    if ($mingw64) { [void]$args.Add("--mingw64") }
+    [void]$args.Add("--standalone")
+    [void]$args.Add("--follow-imports")
+    [void]$args.Add("--output-dir=build")
+    [void]$args.Add("--include-module=wx")
+    [void]$args.Add("--static-libpython=no")
+    [void]$args.Add("--enable-plugin=numpy")
+    [void]$args.Add("--assume-yes-for-downloads")
+    [void]$args.Add("--include-module=glcontext")
+    [void]$args.Add("--include-package-data=*.pyd")
+    [void]$args.Add("--include-data-dir=src/nodes=nodes")
+    [void]$args.Add("--include-plugin-directory=src/nodes")
+    # [void]$installArgs.Add("--windows-icon-from-ico=assets/GIMELSTUDIO_ICO.ico")
```

Comment:
Any reason why this is commented out?

Reply:
It was failing when attempting to do this during the build. I wanted to revisit it when doing the build via automated github action to see if it would work that way, if not, I was going to explore other options for applying the icon after the build is completed. So, I suppose it was mostly kept there as a reminder. What do you recommend, removal for the time being?

Reply:
You can leave it. Maybe add a note there as well, though.